cmake/extensions.cmake

```diff
@@ -3379,7 +3379,7 @@ function(zephyr_linker_dts_memory)
   dt_prop(name PATH ${DTS_MEMORY_PATH} PROPERTY "memory-region")
   if (NOT DEFINED name)
     # Fallback to the node path
-    set(name ${DTS_MEMORY_PATH})
+    set(name \\\"${DTS_MEMORY_PATH}\\\")
```

Comment:
Maybe quoting is better done at the linker script generator itself:
https://github.com/zephyrproject-rtos/zephyr/blob/1f87642f083a7cee0aedad4bc349684ef87b1e71/cmake/linker/ld/ld_script.cmake#L36

Arm Compiler (and future toolchains with custom linkers) could need another way to handle this.

Another alternative could be to sanitize the region names, here using `zephyr_string(SANITIZE name ${name})`
https://github.com/zephyrproject-rtos/zephyr/blob/58234d490c32811860ffc818d662730448c713a7/cmake/extensions.cmake#L2285-L2302

and then similar here:
https://github.com/zephyrproject-rtos/zephyr/blob/1f87642f083a7cee0aedad4bc349684ef87b1e71/include/linker/devicetree_regions.h#L36
sanitize the node path.
@JordanYates any thoughts on this ?

Reply:
This seems like a sane solution, and shouldn't surprise anyone given the same treatment is done with other devicetree properties.

Reply:
It's almost xmas so I took the quickest fix :)